tests/ChatTests/Local.hs

```diff
@@ -0,0 +1,92 @@
+{-# LANGUAGE OverloadedStrings #-}
+{-# LANGUAGE PostfixOperators #-}
+
+module ChatTests.Local where
+
+import ChatClient
+import ChatTests.Utils
+import Test.Hspec
+import System.Directory (copyFile)
+
+chatLocalTests :: SpecWith FilePath
+chatLocalTests = do
+  fdescribe "note folders" $ do
+    it "create folders, add notes, read, search" testNotes
+    it "switch users" testUserNotes
+    it "stores files" testFiles
+
+testNotes :: FilePath -> IO ()
+testNotes tmp = withNewTestChat tmp "alice" aliceProfile $ \alice -> do
+  createFolder alice "self"
+
+  alice ##> "/contacts"
+  -- not a contact
+
+  alice #> "$self keep in mind"
+  alice ##> "/tail"
+  alice <# "$self keep in mind"
+  alice ##> "/chats"
+  alice <# "$self keep in mind"
+  alice ##> "/? keep"
+  alice <# "$self keep in mind"
+
+  alice ##> "/_reaction $1 1 on {\"type\":\"emoji\",\"emoji\":\"🚀\"}"
+  alice <## "added 🚀"
+
+  alice #$> ("/_read chat $1 from=1 to=100", id, "ok")
+  alice ##> "/_unread chat $1 on"
+  alice <## "ok"
+
+  alice ##> "/_delete item $1 1 internal"
+  alice <## "message deleted"
+  alice ##> "/tail"
+  alice ##> "/chats"
+
+  alice #> "$self ahoy!"
+  alice ##> "/_update item $1 1 text Greetings."
+  alice ##> "/tail $self"
+  alice <# "$self Greetings."
+
+  alice ##> "/delete $self"
+  alice <## "note folder self deleted"
+
+testUserNotes :: FilePath -> IO ()
+testUserNotes tmp = withNewTestChat tmp "alice" aliceProfile $ \alice -> do
+  createFolder alice "self"
+
+  alice #> "$self keep in mind"
```

Comment:
re api - we won't allow adding folders, there will only be one folder for notes without names, so it'll be just $